tests/PHPStan/Analyser/LegacyNodeScopeResolverTest.php

```diff
@@ -5129,35 +5129,35 @@ public function dataArrayFunctions(): array
 				'array_values($generalStringKeys)',
 			],
 			[
-				'array<int|non-empty-string, stdClass>',
+				'array<int|non-empty-string, stdClass>&nonEmpty',
 				'array_merge($stringOrIntegerKeys)',
 			],
 			[
-				'array<int|string, DateTimeImmutable|int>',
+				'array<int|string, DateTimeImmutable|int>&nonEmpty',
 				'array_merge($generalStringKeys, $generalDateTimeValues)',
```

Comment:
Im am not confident, whether this is a expected change, or the impl is buggy for this case.